setuptools/dist.py

```diff
@@ -361,6 +357,26 @@ def __init__(self, attrs=None):
                 )
         self._finalize_requires()
 
+    def _install_setup_requires(self, attrs):
+        self.dependency_links = attrs.get('dependency_links')
+        self.setup_requires = attrs.get('setup_requires')
+        # If dependency_links/setup_requires were not passed as
+        # arguments, then we need to honor setup.cfg's options.
+        if (
+            attrs.pop('parse_config', False) and
+            None in (self.dependency_links, self.setup_requires)
+        ):
+            dist = self.__class__()
+            dist.parse_config_files(ignore_option_errors=True)
+            if self.setup_requires is None:
```

Comment:
I think that if both `self.setup_requires` and `dist.setup_requires` are `not None` some warning is required, so that it will alert end-user who accidentally put the option into two places that only one is applied (and two lists are not merged).

Reply:
That would mean always parsing the configuration. What about other options? When passed as arguments to `setup()`, they always override an corresponding configuration in `setup.cfg`, no?

Reply:
what about at least `elif dist.setup_requires is not None:` here?